src/frontend/screens/DownloadManager/index.tsx

```diff
@@ -119,8 +119,12 @@ export default React.memo(function DownloadManager(): JSX.Element | null {
           <div className="dmItemList">
             <DownloadManagerHeader time="queued" />
             {plannendElements.length > 0 ? (
-              plannendElements.map((el, key) => (
-                <DownloadManagerItem key={key} element={el} current={false} />
+              plannendElements.map((el) => (
+                <DownloadManagerItem
+                  key={el.params.appName}
```

Comment:
using the element's appName as the key instead of the list index, so elements are updated properly